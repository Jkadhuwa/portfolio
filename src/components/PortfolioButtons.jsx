import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

function PortfolioButtons({ match: { path } }) {
  const pathname = path.split('/:')[0];

  return (
    <div className="grid gap-y-7 sm:flex justify-center justify-evenly text-base">
      <div>
        <NavLink
          to={`${pathname}`}
          className="shadow-inner text-center border bg-gray-100 rounded py-2 px-4 hover:text-yellow-600 text-gray-500 mr-2 "
          activeClassName="is-active text-yellow-600"
          exact
        >
          SHOW ALL
        </NavLink>
      </div>
      <div className="">
        <NavLink
          to={`${pathname}/webdev`}
          className="shadow-inner text-center border bg-gray-100 rounded py-2 px-2 hover:text-yellow-600 text-gray-500 mr-2"
          activeClassName="is-active text-yellow-600"
          exact
        >
          WEB DEVELOPMENT
        </NavLink>
      </div>
      <div className="">
        <NavLink
          to={`${pathname}/devops`}
          className="shadow-inner text-center border bg-gray-100 rounded py-2 px-2 hover:text-yellow-600 text-gray-500 "
          activeClassName="is-active text-yellow-600"
          exact
        >
          DEVOPS
        </NavLink>
      </div>
    </div>
  );
}
PortfolioButtons.propTypes = {
  match: PropTypes.object,
};

export default PortfolioButtons;
