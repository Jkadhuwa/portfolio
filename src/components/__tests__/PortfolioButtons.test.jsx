import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { StaticRouter } from 'react-router-dom';
import PortfolioButtons from '../PortfolioButtons';

describe('Test PortfolioButton Component', () => {
  afterEach(cleanup);
  it('Should Display Buttons Correctly', async () => {
    const match = {
      path: '/portfolio',
    };
    const { asFragment } = render(
      <StaticRouter>
        <PortfolioButtons match={match} />
      </StaticRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
