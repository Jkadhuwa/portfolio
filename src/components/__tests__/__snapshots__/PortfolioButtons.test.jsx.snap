// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PortfolioButton Component Should Display Buttons Correctly 1`] = `
<DocumentFragment>
  <div
    class="grid gap-y-7 sm:flex justify-center justify-evenly text-base"
  >
    <div>
      <a
        class="shadow-inner text-center border bg-gray-100 rounded py-2 px-4 hover:text-yellow-600 text-gray-500 mr-2 "
        href="/portfolio"
      >
        SHOW ALL
      </a>
    </div>
    <div
      class=""
    >
      <a
        class="shadow-inner text-center border bg-gray-100 rounded py-2 px-2 hover:text-yellow-600 text-gray-500 mr-2"
        href="/portfolio/webdev"
      >
        WEB DEVELOPMENT
      </a>
    </div>
    <div
      class=""
    >
      <a
        class="shadow-inner text-center border bg-gray-100 rounded py-2 px-2 hover:text-yellow-600 text-gray-500 "
        href="/portfolio/devops"
      >
        DEVOPS
      </a>
    </div>
  </div>
</DocumentFragment>
`;
